### PR TITLE
METAL-1061: equinix list of metros

### PR DESCRIPTION
**Pull request by tdomnesc**

### CODE REVIEW

**Comment by tdomnesc** [comment]
/hold
I will unhold Monday

File: pkg/providers/equinix.go [hunk 1]
```diff
@@ -79,22 +79,26 @@ func (p *equinixProvider) Acquire(poolSize int, poolName string, poolType string
...
+			ProjectID: p.config.ProjectID,
+			Tags:      []string{poolName},
+		}
 
-	device, _, err := p.client.Devices.Create(&cr)
-	if err != nil {
-		return resource, fmt.Errorf("error creating device: %w", err)
+		device, _, err := p.client.Devices.Create(&cr)
+		if err != nil {
+			continue
```
Comment (andfasano) [inline pkg/providers/equinix.go:94]: It would be useful to have a log trace here describing which was the error that forced us to skip the current metro

Comment (tdomnesc) [reply pkg/providers/equinix.go:94]: The logger is configured in the cir fsm. I don't know how to bring it inside the provider.

Comment (andfasano) [reply pkg/providers/equinix.go:94]: A `logger` instance could be simply passed through the `providers.NewProvider` (invoked just in `CIResourceFSM.Process` and then stored in the selected factory

**Pull request merged**